src/main/java/com/shade/decima/ui/data/viewer/texture/controls/ImagePanel.java

```diff
@@ -39,7 +68,12 @@ protected void paintComponent(Graphics g) {
             g2.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_NEAREST_NEIGHBOR);
 
             g2.scale(zoom, zoom);
-            g2.drawImage(image, 0, 0, null);
+
+            if (this.channel.equals("RGBA")) {
+                g2.drawImage(image, 0, 0, null);
+            } else {
+                g2.drawImage(createImage(new FilteredImageSource(image.getSource(), new ChannelFilter(this.channel))), 0, 0, null);
```

Comment:
This will result in the image being recreated each time it's rendered.

The better solution would be to recreate it just once in the `setChannel` method. Take a look at the `setMip` method.